src/common.cpp

```diff
@@ -1993,3 +1994,27 @@ void redirect_tty_output() {
     show_stackframe(L'E', 99, 1);
     abort();
 }
+
+/// Return a path to a directory where we can store temporary files.
+wcstring get_path_to_tmp_dir()
+{
+    env_var_t env_tmpdir = env_get_string(L"TMPDIR");
+    if (!env_tmpdir.missing()) return env_tmpdir;
+
+#if defined(_CS_DARWIN_USER_TEMP_DIR)
+    char osx_tmpdir[PATH_MAX];
+    if(confstr(_CS_DARWIN_USER_TEMP_DIR, osx_tmpdir, PATH_MAX - 1))
```

Comment:
If the temp dir is greater than 4096 bytes, this is going to just return a truncated value. Practically speaking that shouldn't happen (I don't know if there's any scenario in which it's possible), but it might be better to handle that here anyway. `confstr` returns the size of buffer needed, so if that size is larger than `PATH_MAX` we could fall back to allocating a dynamic buffer.